### DISCUSSION TITLE
Automatic TOTP Setup

### DISCUSSION THREAD

**Discussion by fullhack2** [mrlt8/docker-wyze-bridge#646] [Q&A]
Hello all,

I had a question regarding using TOTP to automatically generate the required code.

I see that adding the below section in my docker-compose.yml should allow for automatic generation of the code in my authenticator app (if I'm not mistaken and misunderstanding this configuration section)

```
environment:
     - TOTP_KEY=MYSECRETKEY12345
```

I have had trouble getting this enabled on my configuration (seems to generate my container and then just keeps repeating this in the logs):

```
2022/12/19 21:17:14 [WyzeBridge] 🚷 Invalid credentials?
2022/12/19 21:17:14 [WyzeBridge] 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/user/login
2022/12/19 21:18:08 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2022/12/19 21:18:09 [WyzeBridge] 🚷 Invalid credentials?
2022/12/19 21:18:09 [WyzeBridge] 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/user/login
2022/12/19 21:18:29 [WyzeBridge] ☁️ Fetching 'auth' from the Wyze API...
2022/12/19 21:18:29 [WyzeBridge] 🚷 Invalid credentials?
2022/12/19 21:18:29 [WyzeBridge] 400 Client Error: Bad Request for url: https://auth-prod.api.wyze.com/user/login
```

Just to clarify, I should be using the code provided in the initial setup of the authenticator app and provided by the Wyze app as the 'TOTP_KEY'?

Not sure what I'm missing, but it doesn't seem to work for me.  When I manually trigger the 2nd factor by logging into the Wyze app, it accepts the generated code without issue and let's me continue, but wyze-bridge doesn't seem to attempt anything (or give me the option to manually provide the TOTP code via the GUI)

Any thoughts or assistance would be greatly appreciated.

Thanks in advance.

-Fullhack

**Comment by mrlt8** [comment]
Seems like it might be related to #628?
